pkg/doc/signature/suite/jsonwebsignature2020/public_key_verifier.go

```diff
@@ -111,3 +135,58 @@ func parseEllipticCurve(curve string) *ellipticCurve {
 		return nil
 	}
 }
+
+func (v *PublicKeyVerifier) verifyEdDSA(pubKey *sigverifier.PublicKey, signature, msg []byte) error {
+	if pubKey.JWK.Algorithm != "" && pubKey.JWK.Algorithm != "EdDSA" {
+		return fmt.Errorf("unsupported OKP algorithm: %s", pubKey.JWK.Algorithm)
+	}
+
+	// ed25519 panics if key size is wrong
```

Comment:
returns errors.

Reply:
It meant "if we don't check the key size in advance, `ed25519.Verify()` will panic".

Reply:
Changed to "Check the key size before calling ed25519.Verify() as it will panic in case of invalid key size."